### ISSUE TITLE
Bitrate mismatch with firmware 4.36.11.4679

### ISSUE THREAD

**Issue by agrieco** [mrlt8/docker-wyze-bridge#975]
I have a handful of Wyzecams hooked up to wyze-bridge.  I mostly have v3 cameras.

One of my v3 cameras is misbehaving.  The error message is:
[lake-basement] Wrong bitrate=120 does not match 180
[lake-basement] Setting bitrate=180

I noticed this was the only camera I had that was on firmware: 4.36.11.4679 which is odd.   I tried to check the app for firmware upgrades and I don't see any for the other v3 cameras.  Not sure what is going on.

Any ideas?  This doesn't appear to be a benign messge- streaming from this cam is spotty at best.

[WyzeBridge] 🏠 Home Assistant Mode
🚀 DOCKER-WYZE-BRIDGE v2.3.16 
[WyzeBridge] 📚 Using 'auth' from local cache...
 * Serving Flask app 'frontend'
[WyzeBridge] 📚 Using 'user' from local cache...
 * Debug mode: off
[WyzeBridge] 📚 Using 'cameras' from local cache...
[WyzeBridge] 🪄 FILTER ALLOWING: 5 of 15 cams
[WyzeBridge] [+] Adding Lake basement [WYZE_CAKP2JFUS]
[WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://172.30.33.7:5000
[WyzeBridge] Press CTRL+C to quit
[WyzeBridge] [+] Adding Lake side yard [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding Boat [WYZEC1-JZ]
[WyzeBridge] [+] Adding Lake Back Yard [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding Lake front [WYZE_CAKP2JFUS]
[WyzeBridge] starting MediaMTX 1.0.0
[WyzeBridge] 🎬 5 streams enabled
[WyzeBridge] 🎉 Connecting to WyzeCam V3 - Lake side yard on 192.168.10.5
[WyzeBridge] 🎉 Connecting to WyzeCam V2 - Boat on 192.168.10.4
[WyzeBridge] 🎉 Connecting to WyzeCam V3 - Lake Back Yard on 192.168.10.3
[WyzeBridge] 🎉 Connecting to WyzeCam V3 - Lake basement on 192.168.10.2
[lake-side-yard] 📡 Getting 180kb/s HD stream (H264/20fps) via LAN mode (WiFi: 37%) FW: 4.36.10.4054 🔒 (DTLS) (2/3)
[lake-side-yard] WARNING: Skipping smaller frame at start of stream (frame_size=1)
[lake-back-yard] 📡 Getting 180kb/s HD stream (H264/20fps) via LAN mode (WiFi: 41%) FW: 4.36.10.4054 🔒 (DTLS) (2/3)
[lake-back-yard] WARNING: Skipping smaller frame at start of stream (frame_size=1)
[WyzeBridge] ✅ '/lake-side-yard stream is UP! (3/3)
[WyzeBridge] 📖 New client reading from lake-side-yard
[lake-basement] 📡 Getting 180kb/s HD stream (H264/20fps) via LAN mode (WiFi: 40%) FW: 4.36.11.4679 🔒 (DTLS) (2/3)
[WyzeBridge] ✅ '/lake-back-yard stream is UP! (3/3)
[lake-basement] WARNING: Skipping smaller frame at start of stream (frame_size=1)
[WyzeBridge] 📖 New client reading from lake-back-yard
[WyzeBridge] ✅ '/lake-basement stream is UP! (3/3)
[WyzeBridge] 📖 New client reading from lake-basement
[boat] ☁️ Camera is connected via RELAY mode!!
[boat] Stream may consume additional bandwidth!
[boat] 📡 Getting 180kb/s HD stream (H264/15fps) via RELAY mode (WiFi: 38%) FW: 4.9.9.1851 🔒 (DTLS) (2/3)
[WyzeBridge] ✅ '/boat stream is UP! (3/3)
[WyzeBridge] 📖 New client reading from boat
[WyzeBridge] 📕 Client stopped reading from lake-back-yard
[WyzeBridge] 🎉 Connecting to WyzeCam V3 - Lake front on 192.168.10.6
[lake-front] 📡 Getting 180kb/s HD stream (H264/20fps) via LAN mode (WiFi: 49%) FW: 4.36.10.4054 🔒 (DTLS) (2/3)
[lake-front] WARNING: Skipping smaller frame at start of stream (frame_size=1)
[WyzeBridge] ✅ '/lake-front stream is UP! (3/3)
[WyzeBridge] 📖 New client reading from lake-front
[WyzeBridge] 📖 New client reading from lake-back-yard
[lake-basement] Wrong bitrate=120 does not match 180
[lake-basement] Setting bitrate=180
[WyzeBridge] 📕 Client stopped reading from boat
[WyzeBridge] 📕 Client stopped reading from lake-side-yard
[lake-basement] Wrong bitrate=120 does not match 180
[lake-basement] Setting bitrate=180
[WyzeBridge] 📖 New client reading from lake-side-yard
[WyzeBridge] 📖 New client reading from boat
[lake-basement] Wrong bitrate=120 does not match 180
[lake-basement] Setting bitrate=180
[lake-basement] Wrong bitrate=120 does not match 180

**Comment by mrlt8** [comment]
You'll see this if you open the camera stream in the wyze app because the app will request a lower bitrate `120` for HD and `60` for SD. The bridge defaults to using a bitrate of `180` and, since we're pulling the same stream as the app, it will try to maintain the desired bitrate and you'll see that message if the camera switches to a different bitrate. 

You can adjust the desired bitrate that the bridge uses in the [config](https://github.com/mrlt8/docker-wyze-bridge/wiki/Camera-Bitrate-and-Resolution) or by sending a command over the [REST API/MQTT](https://github.com/mrlt8/docker-wyze-bridge/wiki/Camera-Commands).

**Comment by agrieco** [comment]
Thanks for the prompt response.   I opened the app for sure, but the other cameras aren't popping this error (the ones on the older release).   Is there a way to force the camera back to 180 instead of changing the bridge config?

**Comment by Anc0dia** [comment]
I have the same error with my v3 using the same firmware version.  Changing the bitrate in the config doesn't work as it just keeps reporting the bitrate is 120 and wrong as it doesn't match the config value.   

I also can't change the bitrate using API commands, it says successful but nothing changes. 

If I roll back to firmware 4.36.10.4054 I can change the bitrate again.    The issue also exists in the beta firmware 4.36.11.5859.

**Comment by mrlt8** [comment]
The bridge is attempting to change the bitrate when you see `Setting bitrate=180`.

You should be able to get the current bitrate value with the API using:
```
http://<bridge ip>:5000/api/<cam-name>/bitrate
```
And adjust it using:
```
http://<bridge ip>:5000/api/<cam-name>/bitrate/180
```

**Comment by Anc0dia** [comment]
Yes, but it doesn't work with the newer firmware, The bitrate doesn't actually change so it keeps trying over and over.  If you issue the command using API it says successful however it doesn't change the bitrate.     If I roll the firmware back to .4054 it all works as expected.

**Comment by mrlt8** [comment]
Ok, so after a bunch of debugging I think I finally found the issue with the `.11` firmware. 

I mistakenly thought it was the issue was coming from the command to change the bitrate, but it seems to be working as expected and the firmware just isn't sending back the correct bitrate in the `K10020` response #825 which is what we were using to update info about the camera including the bitrate. This is why you'll continually get the wrong bitrate message on 4.36.11.x fw. 

Not sure if this is a bug in the firmware or an intentional change, but we should be able to use `K10050` to lookup the bitrate. One downside is that `K10050` will only return the video parameters so we'll still need to make additional requests to update the rest of info.

**Comment by mrlt8** [comment]
Hmm, I'll have to look into that.

Was it working with 11.4679?

**Comment by Anc0dia** [comment]
I was on 10.4054 and decided to test 11.5859 since it was just released, but since I had the camera down I just tried 11.4679 and it gives the same error.

**Comment by mrlt8** [comment]
The v2.3.17 of the bridge fixed the issue for me on 11.5614. Is the bridge able to set the correct bitrate when it connects to the camera or does it error out right away?

**Comment by Anc0dia** [comment]
I run this in docker on a Debian server so just to eliminate issues there I tried it using it via add-on in Home Assistant & it worked.   So in order for it to work on my Debian machine I need to add this to my compose file:

            - FFMPEG_FLAGS=-avoid_negative_ts make_zero -fflags nobuffer -flags low_delay -strict experimental -fflags +genpts+discardcorrupt+igndts -rw_timeout 5000000 -use_wallclock_as_timestamps 1             

I can then change the bitrate on my v3 camera with the 11.4679 firmware.   I still can't change the bitrate on the v3 pro with firmware 4.58.11.4847

**Comment by mrlt8** [comment]
hmm, I don't think the ffmpeg commands should affect the bitrate stuff since that is coming from a different tutk command. 

We're only checking for the 11.x firmware on the v3 cams, so I'll have to update the script to also use the newer methods for the v3 pro.
